File: Problem1_KLargestElement.py
#Problem 1: https://leetcode.com/problems/kth-largest-element-in-an-array/
#Test Cases passed on Leetcode 
#Time Complexity:    O(n.logk)
#Space Complexity:   O(k)
#Approach:
#create a min heap that contains k maximum elements at any time.
#Remove min element from the heap
#After traversing the entire array,return top element in the heap 
class Solution:
   
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap=[]
        for n in nums:
            heapq.heappush(heap,n)
            if len(heap)>k:
                heapq.heappop(heap)
        return heap[0]

File: Problem1_KLargestElement.java

// Time Complexity : o(nlogk)
// Space Complexity : o(n)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach
class Solution {
    public int findKthLargest(int[] nums, int k) {
        
        if(nums.length==0||nums==null)
            return Integer.MIN_VALUE;
        
        PriorityQueue<Integer> pq=new PriorityQueue<>((a,b)->(a-b));
        
        for(int i:nums)
        {
             pq.add(i);
            
            if(pq.size()>k)
              pq.poll();
        }
           return pq.peek();
    }
}